lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -2,49 +2,79 @@
 import java.util.Arrays;
 import java.util.Iterator;
 import java.util.List;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DynamicTest;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestFactory;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.api.function.ThrowingConsumer;
 import pro.taskana.TaskanaConfiguration;
-import pro.taskana.classification.internal.jobs.ClassificationChangedJob;
-import pro.taskana.common.api.ScheduledJob;
-import pro.taskana.common.internal.jobs.AbstractTaskanaJob;
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.common.api.TaskanaEngine;
+import pro.taskana.common.api.TaskanaEngine.ConnectionManagementMode;
+import pro.taskana.common.api.exceptions.NotAuthorizedException;
+import pro.taskana.common.internal.util.IdGenerator;
 import pro.taskana.common.internal.util.Pair;
-import pro.taskana.common.test.config.DataSourceGenerator;
-import pro.taskana.common.test.security.JaasExtension;
-import pro.taskana.common.test.security.WithAccessId;
+import pro.taskana.simplehistory.impl.SimpleHistoryServiceImpl;
+import pro.taskana.simplehistory.impl.TaskanaHistoryEngineImpl;
 import pro.taskana.simplehistory.impl.jobs.HistoryCleanupJob;
+import pro.taskana.spi.history.api.TaskanaHistory;
 import pro.taskana.spi.history.api.events.task.TaskHistoryEvent;
 import pro.taskana.spi.history.api.events.task.TaskHistoryEventType;
-import pro.taskana.task.internal.jobs.TaskRefreshJob;
-
-@ExtendWith(JaasExtension.class)
-class HistoryCleanupJobAccTest extends AbstractAccTest {
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.models.ObjectReference;
+import pro.taskana.testapi.DefaultTestEntities;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.WithServiceProvider;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.workbasket.api.WorkbasketService;
+
+@WithServiceProvider(
+    serviceProviderInterface = TaskanaHistory.class,
+    serviceProviders = SimpleHistoryServiceImpl.class)
+@TaskanaIntegrationTest
+class HistoryCleanupJobAccTest {
+
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject ClassificationService classificationService;
+  @TaskanaInject TaskanaEngine taskanaEngine;
+
+  ClassificationSummary classification;
+  ObjectReference primaryObjRef;
+  SimpleHistoryServiceImpl historyService;
+  TaskanaConfiguration taskanaConfiguration;
+  TaskanaHistoryEngineImpl taskanaHistoryEngine;
 
+  @WithAccessId(user = "admin")
   @BeforeEach
   void before() throws Exception {
-    resetDb(DataSourceGenerator.getSchemaName());
+    reset();
```

Comment:
I have changed it. I use now nested classes for SimpleHistoryCleanupJobAllCompletedSameParentBusinessIsTrue and SimpleHistoryCleanupJobAllCompletedSameParentBusinessIsFalse

So the property will be changed accordingly through the modify function, so that only the taskanaEngine and the services at that class will be effected